### ISSUE TITLE
Term incorrectly marked as obsolete when reading go-basic.json

### ISSUE THREAD

**Issue by dhimmel** [althonos/pronto#122]
```python
>>> import pronto
# GO:0000003 is obsolete when reading go-basic.json
>>> url = "http://release.geneontology.org/2021-02-01/ontology/go-basic.json.gz"
>>> go = pronto.Ontology(handle=url)
>>> go["GO:0000003"].obsolete
True

# GO:0000003 is NOT obsolete when reading go-basic.obo
>>> url = "http://release.geneontology.org/2021-02-01/ontology/go-basic.obo"
>>> go = pronto.Ontology(handle=url)
>>> go["GO:0000003"].obsolete
False

# error when reading go-basic.owl
>>> url = "http://release.geneontology.org/2021-02-01/ontology/go-basic.owl"
>>> go = pronto.Ontology(handle=url)
>>> go["GO:0000003"].obsolete
KeyError                                  Traceback (most recent call last)
     83                 for superentity in lineage.sup:
---> 84                     graphdata.lineage[superentity].sub.add(subentity)
     85 
     86     def import_lineage(self):
KeyError: 'RO:0002211'
```

Looking at the `go-basic.json` source, I don't see anything indicating this node is obsolete:

```bash
curl --silent http://release.geneontology.org/2021-02-01/ontology/go-basic.json.gz \
  | zcat | grep --after-context=40 --before-context=1 \
  '"id" : "http://purl.obolibrary.org/obo/GO_0000003"'
```

```json
    }, {
      "id" : "http://purl.obolibrary.org/obo/GO_0000003",
      "meta" : {
        "definition" : {
          "val" : "The production of new individuals that contain some portion of genetic material inherited from one or more parent organisms.",
          "xrefs" : [ "GOC:go_curators", "GOC:isa_complete", "GOC:jl", "ISBN:0198506732" ]
        },
        "subsets" : [ "http://purl.obolibrary.org/obo/go#goslim_plant", "http://purl.obolibrary.org/obo/go#goslim_pir", "http://purl.obolibrary.org/obo/go#goslim_flybase_ribbon", "http://purl.obolibrary.org/obo/go#goslim_chembl", "http://purl.obolibrary.org/obo/go#goslim_generic", "http://purl.obolibrary.org/obo/go#goslim_agr" ],
        "xrefs" : [ {
          "val" : "Wikipedia:Reproduction"
        } ],
        "synonyms" : [ {
          "pred" : "hasExactSynonym",
          "val" : "reproductive physiological process",
          "xrefs" : [ ]
        } ],
        "basicPropertyValues" : [ {
          "pred" : "http://www.geneontology.org/formats/oboInOwl#hasOBONamespace",
          "val" : "biological_process"
        }, {
          "pred" : "http://www.geneontology.org/formats/oboInOwl#hasAlternativeId",
          "val" : "GO:0019952"
        }, {
          "pred" : "http://www.geneontology.org/formats/oboInOwl#hasAlternativeId",
          "val" : "GO:0050876"
        } ]
      },
      "type" : "CLASS",
      "lbl" : "reproduction"
    }, {
```

Could there be an issue somewhere in the JSON OBO Graph importer?

Regarding `KeyError: 'RO:0002211'`, would it make sense to log a warning but continue reading?

**Comment by althonos** [comment]
@dhimmel : I think this is fixed with `v2.4.0`, at least on my end I get:
```python
>>> import pronto
>>> go = pronto.Ontology("http://release.geneontology.org/2021-02-01/ontology/go-basic.json.gz")
>>> go['GO:0000003']
Term('GO:0000003', name='reproduction')
>>> go['GO:0000003'].obsolete
False
```

**Comment by althonos** [comment]
`go-basic.owl` parses successfully in `v2.4.1`.

**Comment by dhimmel** [comment]
Thanks @althonos!

I'm curious to learn what change in 2.4.0 or 2.4.1 fixed this issue. Is there a specific commit you're aware of that fixes this or just that this issue no longer occurs emperically?

**Comment by althonos** [comment]
So the first issue (OBO/OBO-JSON) was likely related to the `alt_id` short-cutting requested in #100 and implemented in `v2.3.0`. Since it was causing a lot of bugs (#120, #126) I removed it in `v2.4.0`. This was enough to fix the issue here.

The second issue (OWL not parsing) was fixed in 2add50042d4e39e697543728c63b974ca8c8c90b and 96cb752b08c7427725238b9f900b26d311ac81cb. As you know, the RDF/XML parser will perform URI compaction using the semantics from the OBO guide, so the following object property in OWL:
```xml
<owl:ObjectProperty rdf:about="http://purl.obolibrary.org/obo/RO_0002211">
    <rdf:type rdf:resource="http://www.w3.org/2002/07/owl#TransitiveProperty"/>
    <oboInOwl:hasDbXref rdf:datatype="http://www.w3.org/2001/XMLSchema#string">RO:0002211</oboInOwl:hasDbXref>
    <oboInOwl:id rdf:datatype="http://www.w3.org/2001/XMLSchema#string">regulates</oboInOwl:id>
    <oboInOwl:shorthand rdf:datatype="http://www.w3.org/2001/XMLSchema#string">regulates</oboInOwl:shorthand>
    <rdfs:label rdf:datatype="http://www.w3.org/2001/XMLSchema#string">regulates</rdfs:label>
</owl:ObjectProperty>
```
gets translated to the following OBO typedef:
```ini
[Typedef]
id: regulates
name: regulates
xref: RO:0002211
is_transitive: true
```
using the `oboInOwl:id` or the `oboInOwl:shorthand` to replace the URI with a CURIE.

However, this was not done consistently, and in the case a property was declared as a subproperty of an object property that had its URI compacted, the parser didn't use the CURIE, and tried to look for a non-existing entity, raising a `KeyError`. 

In `go-basic.owl`, this was caused by:
```xml
<owl:ObjectProperty rdf:about="http://purl.obolibrary.org/obo/RO_0002212">
    <rdfs:subPropertyOf rdf:resource="http://purl.obolibrary.org/obo/RO_0002211"/>
    <oboInOwl:hasDbXref rdf:datatype="http://www.w3.org/2001/XMLSchema#string">RO:0002212</oboInOwl:hasDbXref>
    <oboInOwl:id rdf:datatype="http://www.w3.org/2001/XMLSchema#string">negatively_regulates</oboInOwl:id>
    <oboInOwl:shorthand rdf:datatype="http://www.w3.org/2001/XMLSchema#string">negatively_regulates</oboInOwl:shorthand>
    <rdfs:label rdf:datatype="http://www.w3.org/2001/XMLSchema#string">negatively regulates</rdfs:label>
</owl:ObjectProperty>
```
which inherits from `RO:0002211`; however, `RO:0002211` was previously renamed `regulates`, so the parser was not able to add `RO:0002211` as a superclass.
